src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -531,4 +533,226 @@ private double convertInch1200ToPixels(double inch1200Value){
+        List<String> illustrationList = new ArrayList<>();
+        List<IllustrationMetadata> metadataList = new ArrayList<>();
+
+
+        // Extract metadata from SolrDocument
+        String pageUUID = doc.getFieldValue("pageUUID").toString();
+        String correctUUID = convertPageUUID(pageUUID);
+        if (!uniqueUUIDs.add(correctUUID)){
+            return null;
+        }
+        long pageWidth = (long) doc.getFieldValue("page_width");
+        long pageHeight = (long) doc.getFieldValue("page_height");
+        List<String> illustrations = (List<String>) doc.getFieldValue("illustration");
+        // Check if illustrations are present. If not, continue to next SolrDocument in list
+        if (illustrations == null) {
+            return Stream.empty();
+        }
+
+        return illustrations.stream().map(metadata -> new IllustrationMetadata(metadata, correctUUID, pageWidth, pageHeight));
+    }
+
+    /**
+     * Streams content of all metadatas given in input stream to an output stream that delivers a zip file of all images from the URLs.
+     * @param illustrationMetadata used to stream URLS from and construct filenames.
+     * @param output stream which holds the outputted zip file.
+     * @param metadataMap which delivers overall information on the export.
+     * @param exportFormat determines what kind of export that are to be done.
+     */
+    public void streamUrls(Stream<? extends BasicMetadata> illustrationMetadata, OutputStream output, Map<String, Object> metadataMap, String exportFormat) throws IOException {
+        ZipOutputStream zos = new ZipOutputStream(output);
+        zos.setLevel(Deflater.NO_COMPRESSION);
+
+        // Add metadata file to zip
+        try {
+            addMetadataFileToZip(metadataMap, zos);
+        } catch (Exception e) {
+            String message = String.format(
+                    Locale.ROOT,
+                    "Exception adding metadata entry to ZIP with  %s illustrationMetadatas",
+                    illustrationMetadata == null ? "null" : illustrationMetadata.count());
+            log.warn(message, e);
+            throw new IOException(message, e);
+        }
+
+        AtomicInteger count = new AtomicInteger();
+        try {
+            switch (exportFormat){
+                case "illustrations":
+                    illustrationMetadata.map(metadata -> (IllustrationMetadata) metadata)
+                            .forEach(metadata -> exportIllustrations(metadata, exportFormat, count, zos));
+                    zos.close();
+                    break;
+                case "fullPage":
+                    illustrationMetadata.map(metadata -> (FullPageMetadata) metadata)
+                            .forEach(metadata -> exportFullpage(metadata, exportFormat, count, zos));
+                    zos.close();
+                    break;
```

Comment:
This is duplicated code :-)

But we talked about this and cleaning that will be another issue.